packages/arb-bridge-eth/contracts/rollup/Rollup.sol

```diff
@@ -0,0 +1,699 @@
+     * @param executionCheckTime Amount of time to check the assertion's execution
+     */
+    function createChallenge(
+        address payable[2] calldata stakers,
+        uint256[2] calldata nodeNums,
+        bytes32[3] calldata nodeFields,
+        uint256 executionCheckTime
+    ) external whenNotPaused {
+        require(nodeNums[0] < nodeNums[1], "WRONG_ORDER");
+        require(nodeNums[1] <= latestNodeCreated(), "NOT_PROPOSED");
+        require(latestConfirmed() < nodeNums[0], "ALREADY_CONFIRMED");
+
+        INode node1 = getNode(nodeNums[0]);
+        INode node2 = getNode(nodeNums[1]);
+
+        require(node1.prev() == node2.prev(), "DIFF_PREV");
+
+        checkUnchallengedStaker(stakers[0]);
+        checkUnchallengedStaker(stakers[1]);
+
+        require(node1.stakers(stakers[0]), "STAKER1_NOT_STAKED");
+        require(node2.stakers(stakers[1]), "STAKER2_NOT_STAKED");
+
+        require(
+            node1.challengeHash() ==
+                ChallengeLib.challengeRootHash(
+                    nodeFields[0],
+                    nodeFields[1],
+                    nodeFields[2],
+                    executionCheckTime
+                ),
+            "CHAL_HASH"
+        );
+
+        // Start a challenge between staker1 and staker2. Staker1 will defend the correctness of node1, and staker2 will challenge it.
+        address challengeAddress =
+            challengeFactory.createChallenge(
+                address(this),
+                nodeFields[0],
+                nodeFields[1],
+                nodeFields[2],
+                executionCheckTime,
+                stakers[0],
+                stakers[1],
+                challengePeriodBlocks
+            );
+
+        challengeStarted(stakers[0], stakers[1], challengeAddress);
+
+        emit RollupChallengeStarted(challengeAddress, stakers[0], stakers[1], nodeNums[0]);
+    }
+
+    /**
+     * @notice Inform the rollup that the challenge between the given stakers is completed
+     * @dev completeChallenge isn't pausable since in flight challenges should be allowed to complete or else they could be forced to timeout
+     * @param winningStaker Address of the winning staker
+     * @param losingStaker Address of the losing staker
+     */
+    function completeChallenge(address winningStaker, address losingStaker) external override {
+        // Only the challenge contract can declare winners and losers
+        require(msg.sender == inChallenge(winningStaker, losingStaker), "WRONG_SENDER");
+
+        uint256 loserStake = amountStaked(losingStaker);
+        uint256 winnerStake = amountStaked(winningStaker);
+        if (loserStake > winnerStake) {
+            loserStake -= reduceStakeTo(losingStaker, winnerStake);
+        }
+
+        uint256 amountWon = loserStake / 2;
+        increaseStakeBy(winningStaker, amountWon);
+        loserStake -= amountWon;
+        clearChallenge(winningStaker);
+
+        // TODO: deposit extra loserStake into ArbOS
```

Comment:
You can give "burned" funds to ArbOS by doing a payable call to ArbOwner.addToReserveFunds(), with the chain owner as the sender.

Reply:
This is gonna be way more complicated because we support staking in tokens, but the main bridge only support ETH so this will take more work

Reply:
The best way is probably just to deposit the burned tokens to some configurable address at L2.  Then we can create an L2 contract later that disposes of them in some useful way.